server/mlabns/tests/test_resolver.py

```diff
@@ -78,6 +89,8 @@ def testGetCandidatesYesMemcache(self):
         self.testbed = testbed.Testbed()
         self.testbed.activate()
         self.testbed.init_memcache_stub()
+        _apply_namespace_bug_workaround()
```

Comment:
The `testGetCandidatesYesMemcache` function contains just one test. Before every test in this module that depends on the datastore, we have to create and initialize a testbed (see https://cloud.google.com/appengine/docs/python/tools/localunittesting) and we need to apply the namespace workaround after any activation of the test bed.

I'm trying to keep these changes as small as possible because my first priority is getting the code in a stable state where the tests work. I *very* much agree with you that lots of this module needs to be refactored, but I'd like to hold off on any refactoring work until a later CL. It's hard to refactor until we have all tests passing because we can't tell if the refactor maintained correct behavior.